pkg/csi_driver/controller.go

```diff
@@ -136,6 +137,12 @@ func (s *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVolu
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
+	volumeID := getVolumeIDFromFileInstance(newFiler, modeInstance)
```

Comment:
Why are we doing this after checking for a snapshot source? I'd think we'd want to block before collecting such info so that there's less of a window for a race, eg a snapshot getting deleted.

Reply:
Moved it